File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int,int> conjugate_map; //Stores indexes

        for(int i=0; i<nums.size(); i++){
            if(conjugate_map.count(target - nums[i])){
                return {conjugate_map[target - nums[i]],i};
            }
            conjugate_map[nums[i]] = i;
        }
        
        return {};
        
    }
};

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complement_map = collections.defaultdict(int)
        for i in range(len(nums)):
            if target-nums[i] in complement_map:
                return [complement_map[target-nums[i]],i]
            complement_map[nums[i]] = i
        return []
                
        